Add Gumbel distribution

maxliving opened the pull request:
I'm pretty unfamiliar with this codebase, so it's possible I've done this incorrectly. Getting some errors in `test_variational_inference.py` (using Docker for testing, as described in `CONTRIBUTING.md`), but I think they're unrelated to this change.

Commit:
Add Gumbel distribution

twiecki commented on pymc3/distributions/continuous.py:
```diff
@@ -1344,5 +1344,49 @@ def logp(self, value):
                          tt.switch(tt.eq(value, c), tt.log(2 / (upper - lower)),
-                         tt.switch(alltrue_elemwise([c < value, value <= upper]),
-                         tt.log(2 * (upper - value) / ((upper - lower) * (upper - c))),np.inf)))
+                                   tt.switch(alltrue_elemwise([c < value, value <= upper]),
```
these lines shouldn't change.

twiecki commented on pymc3/distributions/continuous.py:
```diff
@@ -1344,5 +1344,49 @@ def logp(self, value):
+
+    def logp(self, value):
+        scaled = (value - self.mu) / self.beta
+        return -scaled - tt.exp(-scaled) - tt.log(self.beta)
```
need `bound` for self.beta if it has only positive support.

twiecki commented:
See minor comments, otherwise, this looks great!

Commit:
Bound Gumbel logp to handle negative beta

maxliving commented:
@twiecki Thanks for the quick review! Addressed your comments. Also do you have any thoughts on why the unit tests might be failing when run locally (in Docker) but not here?

twiecki commented on pymc3/tests/test_distributions.py:
```diff
@@ -1,6 +1,8 @@
 
 import itertools
+
+from pymc3.distributions.continuous import Gumbel
```
this should move to the other imports

maxliving replied:
Ahh, damn Pycharm. Done.

Commit:
Fix Gumbel import in unit test

maxliving commented:
@twiecki OK to merge?

twiecki commented:
Thanks @maxliving!
